src/ParallelRun.jl

```diff
@@ -296,8 +299,388 @@ function load_images(rcfs, stagedir)
+
+Returns:
+
+- Dictionary of results, keyed by SDSS thing_id.
+"""
+function one_node_infer_multi_iter(rcfs::Vector{RunCamcolField},
+                                   stagedir::String;
+                                   cyclades_partition=true,
+                                   n_iters=10,
+                                   objid="",
+                                   box=BoundingBox(-1000., 1000., -1000., 1000.),
+                                   primary_initialization=true,
+                                   reserve_thread=Ref(false),
+                                   thread_fun=phalse,
+                                   timing=InferTiming())
+    nprocthreads = nthreads()
+    if reserve_thread[]
+        nprocthreads = nprocthreads-1
+    end
+    Log.info("Running with $(nprocthreads) threads")
+    
+    # Read all primary objects in these fields.
+    tic()
+    duplicate_policy = primary_initialization ? :primary : :first
+    catalog = SDSSIO.read_photoobj_files(rcfs, stagedir,
+                        duplicate_policy=duplicate_policy)
+    timing.read_photoobj = toq()
+    Log.info("$(length(catalog)) primary sources")
+
+    reserve_thread[] && thread_fun(reserve_thread)
+
+    # Filter out low-flux objects in the catalog.
+    catalog = filter(entry->(maximum(entry.star_fluxes) >= MIN_FLUX), catalog)
+    Log.info("$(length(catalog)) primary sources after MIN_FLUX cut")
+
+    # Filter any object not specified, if an objid is specified
+    if objid != ""
+        Log.info(catalog[1].objid)
+        catalog = filter(entry->(entry.objid == objid), catalog)
+    end
+
+    # Get indicies of entries in the  RA/Dec range of interest.
+    entry_in_range = entry->((box.ramin < entry.pos[1] < box.ramax) &&
+                             (box.decmin < entry.pos[2] < box.decmax))
+    target_sources = find(entry_in_range, catalog)
+
+    nputs(dt_nodeid, string("processing $(length(target_sources)) sources in ",
+          "$(box.ramin), $(box.ramax), $(box.decmin), $(box.decmax)"))
+
+    # If there are no objects of interest, return early.
+    if length(target_sources) == 0
+        return Dict{Int, Dict}()
+    end
+
+    reserve_thread[] && thread_fun(reserve_thread)
+
+    # Read in images for all (run, camcol, field).
+    tic()
+
+    images = load_images(rcfs, stagedir)
+    timing.read_img = toq()
+
+    reserve_thread[] && thread_fun(reserve_thread)
+
+    Log.info("Finding neighbors")
+    tic()
+    neighbor_map = Infer.find_neighbors(target_sources, catalog, images)
+    Log.info("Neighbors found in $(toq()) seconds")
+
+    reserve_thread[] && thread_fun(reserve_thread)
+
+    # Partition the sources
+    n_sources = length(target_sources)
+    Log.info("Optimizing $(n_sources) sources")
+    if cyclades_partition
+        # Convert neighbormap to cyclades map (map from source_id -> [neighbor source_ids])
+        cyclades_neighbor_map = Dict{Int64, Vector{Int64}}()
```

Comment:
Hm, it seems like doing the partitioning is relatively very fast compared to allocation and optimization (0.000236101 seconds vs 209.690328562s for 104 sources). I'm thinking maybe we won't need to parallelize the connected components code for single node because of this. What do you think?